temporalcli/commands.workflow_reset_test.go

```diff
@@ -248,27 +248,112 @@ func (s *SharedServerSuite) TestBatchResetByBuildId() {
 	sut.stopWorkerFor("v3")
 }
 
-func (s *SharedServerSuite) TestWorkflow_Reset_ReapplyExclude() {
-	var wfExecutions, timesSignalSeen int
+type WorkflowResetTest struct {
+	s                      *SharedServerSuite
+	reapplyType            string
+	reapplyExclude         []string
+	expectUpdatesReapplied bool
+	expectSignalsReapplied bool
+}
+
+func (s *SharedServerSuite) TestWorkflow_Reset_DefaultReappliesAll() {
+	t := WorkflowResetTest{
+		s:                      s,
+		expectUpdatesReapplied: true,
+		expectSignalsReapplied: true,
+	}
+	t.run()
+}
+
+func (s *SharedServerSuite) TestWorkflow_Reset_ExcludeUpdate() {
+	t := WorkflowResetTest{
+		s:                      s,
+		reapplyExclude:         []string{"Update"},
+		expectUpdatesReapplied: false,
+		expectSignalsReapplied: true,
+	}
+	t.run()
+}
+
+func (s *SharedServerSuite) TestWorkflow_Reset_ExcludeSignal() {
+	t := WorkflowResetTest{
+		s:                      s,
+		reapplyExclude:         []string{"Signal"},
+		expectUpdatesReapplied: true,
+		expectSignalsReapplied: false,
+	}
+	t.run()
+}
+
+func (s *SharedServerSuite) TestWorkflow_Reset_ExcludeSignalAndUpdate() {
+	t := WorkflowResetTest{
+		s:                      s,
+		reapplyExclude:         []string{"Signal", "Update"},
+		expectUpdatesReapplied: false,
+		expectSignalsReapplied: false,
+	}
+	t.run()
+}
+
+func (s *SharedServerSuite) TestWorkflow_Reset_ReapplySignalOnly() {
+	t := WorkflowResetTest{
+		s:                      s,
+		reapplyType:            "Signal",
+		expectUpdatesReapplied: false,
+		expectSignalsReapplied: true,
+	}
+	t.run()
+}
+
+func (t *WorkflowResetTest) run() {
+	s := t.s
+	var wfExecutions, updateHandlerExecutions, signalHandlerExecutions int
 	s.Worker().OnDevWorkflow(func(ctx workflow.Context, a any) (any, error) {
-		sigChan := workflow.GetSignalChannel(ctx, "sig")
+		// Handle signals
+		sigChan := workflow.GetSignalChannel(ctx, "mySignal")
 		workflow.Go(ctx, func(ctx workflow.Context) {
 			for {
 				sigChan.Receive(ctx, nil)
-				fmt.Println("saw signal", workflow.GetInfo(ctx).WorkflowExecution)
-				timesSignalSeen++
+				signalHandlerExecutions++
 			}
 		})
-		err := workflow.Sleep(ctx, 1*time.Second)
+		// Handle updates
+		workflow.SetUpdateHandler(ctx, "myUpdate", func(ctx workflow.Context) error {
+			updateHandlerExecutions++
+			return nil
+		})
+		err := workflow.Sleep(ctx, 100*time.Millisecond)
```

Comment:
This test could be written without sleep. In fact, it would be sufficient to just check that the CLI is mapped correctly to gRPC requests -- reset itself is tested in the SDK and server. But that's all orthogonal to this PR.

Reply:
So long as not flaky, no concerns here